apps/ui/src/views/Proposal/Overview.vue

```diff
@@ -150,6 +153,32 @@ async function handleAiSummaryClick() {
     aiSummaryLoading.value = false;
   }
 }
+
+async function handleAiSpeechClick() {
+  if (aiSpeechAudio.value) {
+    return audioState.value === 'playing' ? pause() : play();
+  }
+
+  try {
+    aiSpeechLoading.value = true;
+    const response = await fetch(`https://sh5.co/api/ai/tts/${props.proposal.id}`, {
+      method: 'POST'
+    });
+
+    const audioBuffer = await response.arrayBuffer();
+    await init(audioBuffer);
```

Comment:
Shouldn't we have this logic in the composable file? Ideally we want keep the .vue files light

Reply:
Seems it was fine to have it there from https://github.com/snapshot-labs/sx-monorepo/pull/193#pullrequestreview-1960098038

Reply:
Or maybe we can have a single composable to handle both summary and speech ?

Reply:
Yes maybe, one specific for sidekick stuffs